pennylane/registers.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the :func:`registers` function.
+"""
+
+from .wires import Wires
+
+
+def registers(register_dict):
+    """
+    This function helps manage wire registers when setting up a quantum algorithm.
+    Wire register names and their total number of wires are typically known in advance,
+    but managing the specific wire range for each register can be a challenge. The
+    ``qml.registers()`` function helps by creating a dictionary, mapping from register name to
+    wire range. Moreover, it is possible to input a nested structure where registers
+    contain sub-registers, as illustrated in the example below.
+
+    Args:
+        register_dict (dict): a dictionary where keys are register names and values are either
+            positive integers indicating the number of qubits or nested dictionaries of more registers.
```

Comment:
Why couldn't they be of type `Wires`?

Reply:
Hey @maliasadi! Utkarsh had a similar concern. 

Wires instances being accepted poses a question for how other wires objects should be ordered. E.g., how do we expect to index / label wires in this example?

`qml.registers({"reg1": 3, "reg2": qml.wires.Wires(["A", 1, 42]), ...})`

It would also require an additional search through the entire dictionary, an additional complexity overhead that, yes, may be peanuts now 🥜, but might not be with millions of qubits 📈!

Another weird example of mixing `registers` with already-instantiated wires:

```
wires_dict = qml.registers({"reg1": 3, "reg2": 2})
wires_dict["new_reg"] = already_instantiated_wires
```

But `already_instantiated_wires` might have overlapping wire labels (might be unintentional!)

Overall, I think this needs a bit more thought and care. If this gets requested by users, we will certainly act on it if possible!

Reply:
Ah, I see! Thanks for the explanation! 

PS. I couldn't find Utkarsh comment ;)